dspy/teleprompt/finetune_teleprompter.py

```diff
@@ -0,0 +1,237 @@
+from typing import Any, Callable, Dict, List, Optional, Union
+
+import dspy
+from dspy.utils.logging import logger
+from dspy.evaluate.evaluate import Evaluate
+from dspy.primitives.example import Example
+from dspy.primitives.program import Program
+from dspy.primitives.prediction import Prediction
+
+_INFO_DEFAULT_TEACHER = """No teacher provided. Using a copy of the student \
```

Comment:
These messages are only used once, we can inline them

Reply:
Done!